### ISSUE TITLE
Invalid position given by 'undefined'

### ISSUE THREAD

**Issue by CraigFan96** [AtomLinter/linter-eslint#847]
ESLint returned a point that did not exist in the document being edited.
Rule: `undefined`
Requested range: 16:1

Debug information:
```json
{
  "atomVersion": "1.14.2",
  "linterEslintVersion": "8.1.2",
  "linterEslintConfig": {
    "lintHtmlFiles": false,
    "useGlobalEslint": false,
    "showRuleIdInMessage": true,
    "disableWhenNoEslintConfig": true,
    "eslintrcPath": "",
    "globalNodePath": "",
    "advancedLocalNodeModules": "",
    "eslintRulesDir": "",
    "disableEslintIgnore": false,
    "disableFSCache": false,
    "fixOnSave": false,
    "scopes": [
      "source.js",
      "source.jsx",
      "source.js.jsx",
      "source.babel",
      "source.js-semantic"
    ],
    "rulesToSilenceWhileTyping": [],
    "rulesToDisableWhileFixing": []
  },
  "eslintVersion": "1.10.3",
  "hoursSinceRestart": 9.8,
  "platform": "darwin",
  "eslintType": "local project",
  "eslintPath": "/Users/CraigFan/Documents/CS326/Workshop6/node_modules/eslint"
}
```

**Comment by Arcanemagus** [comment]
Do you have a file and configuration that you can share that reproduces this? The only other time this was reported the person never responded and we were unable to determine how they got to that state.

**Comment by colepeters** [comment]
I got this issue as well, and it turned out to be a cold old fashioned case of some missing braces and parentheses.

This was generated from Atom’s report:

ESLint returned a point that did not exist in the document being edited.
Rule: `undefined`
Requested range: 17:1

Debug information:
```json
{
  "atomVersion": "1.15.0",
  "linterEslintVersion": "8.1.4",
  "linterEslintConfig": {
    "lintHtmlFiles": false,
    "useGlobalEslint": false,
    "showRuleIdInMessage": true,
    "disableWhenNoEslintConfig": true,
    "eslintrcPath": "",
    "globalNodePath": "",
    "advancedLocalNodeModules": "",
    "eslintRulesDir": "",
    "disableEslintIgnore": false,
    "disableFSCache": false,
    "fixOnSave": false,
    "scopes": [
      "source.js",
      "source.jsx",
      "source.js.jsx",
      "source.babel",
      "source.js-semantic"
    ],
    "rulesToSilenceWhileTyping": [],
    "rulesToDisableWhileFixing": []
  },
  "eslintVersion": "1.10.3",
  "hoursSinceRestart": 0.1,
  "platform": "darwin",
  "eslintType": "local project",
  "eslintPath": "/Users/cole/Sites/gargantua/node_modules/eslint",
  "editorScopes": [
    "source.js.jsx"
  ]
}
```

…which appeared while linting this file:

```js
import isHexColor from '../../../src/js/utils/isHexColor';

const validHexes = [ 'f00', 'ff0000', '0f0', 'abcdef' ];
const invalidHexes = [ 'f0', 'ff00000', 'red', 'l0lwat' ];

describe('Given the `isHexColor` helper function', function () {
  describe('when it is passed a valid hex color', function () {
    it('should return `true`', function () {
      validHexes.forEach(color => expect(isHexColor(color)).to.be.true);
    });
  });
  describe('when it is passed a invalid hex color', function () {
    it('should return `false`', function () {
      invalidHexes.forEach(color => expect(isHexColor(color)).to.be.false);
    });
});
```

…which I later realised was missing a `});`. After fixing that, the error was resolved. 😄

**Comment by colepeters** [comment]
(For what it's worth, it would've been cool if the linter told me about my missing `});` rather than a mysterious `Invalid position given…`.) ;)

**Comment by IanVS** [comment]
So, my guess is that it was a parsing error, and was returning an invalid position.  You are on a very old version of ESLint.  I wonder if you would have gotten a better message if you were on something more recent.  When I get some time I'll try to reproduce.

**Comment by Arcanemagus** [comment]
As @IanVS said, this is most likely caused by some long fixed bug in ESLint as you are both using the same ancient version. Could one of you post your ESLint configuration so I can attempt to reproduce this fully?

**Comment by fuunnx** [comment]
Hi, I have this issue as well:

ESLint returned a point that did not exist in the document being edited.
Rule: `undefined`
Requested range: 24:1



My code
```javascript
import seedrandom from 'seed-random'

const seed = ''
const randomDriver = RandomDriver(seed)

export default function makeRandomDriver (seed1) {
//...
}
export makeRandomDriver // when I remove this line, it all works !

```

Debug information:
```json
{
  "atomVersion": "1.15.0",
  "linterEslintVersion": "8.1.2",
  "linterEslintConfig": {
    "lintHtmlFiles": false,
    "useGlobalEslint": false,
    "showRuleIdInMessage": true,
    "disableWhenNoEslintConfig": true,
    "eslintrcPath": "",
    "globalNodePath": "",
    "advancedLocalNodeModules": "",
    "eslintRulesDir": "",
    "disableEslintIgnore": false,
    "disableFSCache": false,
    "fixOnSave": false,
    "scopes": [
      "source.js",
      "source.jsx",
      "source.js.jsx",
      "source.babel",
      "source.js-semantic"
    ],
    "rulesToSilenceWhileTyping": [],
    "rulesToDisableWhileFixing": []
  },
  "eslintVersion": "1.10.3",
  "hoursSinceRestart": 3.5,
  "platform": "darwin",
  "eslintType": "local project",
  "eslintPath": "/Users/Fuunnx/workspace/funx/2016/funx.fr/node_modules/eslint"
}
```

=)

**Comment by IanVS** [comment]
Everyone so far posting a debug message in this issue is using version `1.10.3` of ESLint, which is very old.  Can anyone verify this problem is resolved when updating to a more recent version of ESLint?

**Comment by fuunnx** [comment]
Yep, I uptaded to Eslint 3.18.0 and everything is fine =)

Thanks for maintaining this package 🤘

**Comment by Arcanemagus** [comment]
I'm still somewhat curious how all of you managed to be using the exact same version of ESLint, especially since it is so old.

**Comment by IanVS** [comment]
@Arcanemagus I believe it was the final version in the `1.x` range.
